File: service/src/main/java/com/db/service/state/ActiveStudentState.java
package com.db.service.state;

import org.springframework.stereotype.Component;

@Component
public class ActiveStudentState implements StudentState {
    @Override
    public boolean canRegisterCourse() {
        return true;
    }

    @Override
    public String getStatusName() {
        return "ACTIVE";
    }
}

File: ui/src/app/core/state/impl/active-student.state.ts
import { Injectable } from '@angular/core';
import { StudentState } from '../student-state';

@Injectable({ providedIn: 'root' })
export class ActiveStudentState implements StudentState {
  getStatusName(): string { return 'ACTIVE'; }
  canEdit(): boolean { return true; }
  canRegisterCourse(): boolean { return true; }
  canRestore(): boolean { return false; }
}
